File: src/ruby/example-lib/jruby_bridge.rb
require 'java'

# This class is a bridge between ruby code and the java interface
# `com.example.JRubyBridge`.

module Example
  class JRubyBridge
    include Java::com.example.JRubyBridge

    def initialize
      puts "==> JRubyBridge#initialize called in Ruby"
    end

    # This rubyVersion method is part of the JRubyBridge java interface
    def rubyVersion
      # Just return the version of ruby
      RUBY_VERSION
    end
  end
end

File: src/java/com/example/JRubyBridge.java
package com.example;

/**
 * This interface is a bridge between Clojure/Java code and the ruby class
 * `JRubyBridge` (see src/ruby/example-lib/jruby_bridge.rb).
 *
 * The ruby class uses JRuby to "implement" this interface.
 * From Clojure or Java, interact with an instance of the ruby class as if it
 * were an isntance of this interface.
 */
public interface JRubyBridge {
  String rubyVersion();
}
